From: Miklos Vajna <vmiklos@frugalware.org>
Date: Sat, 28 Feb 2009 22:46:57 +0000 (+0100)
Subject: update-ignore.d/01update-ignore: add support for mktemp
X-Git-Url: http://git.kitenet.net/?p=etckeeper.git;a=commitdiff_plain;h=3b362a9abf4fdbe3d2103b2aa13c86b8c2613726

update-ignore.d/01update-ignore: add support for mktemp

On system where tempname is not installed, check if we could use mktemp
instead.

If we are at it, add a warning if neither of them is available.

Signed-off-by: Miklos Vajna <vmiklos@frugalware.org>
---

diff --git a/update-ignore.d/01update-ignore b/update-ignore.d/01update-ignore
index 8156437..510e13f 100755
--- a/update-ignore.d/01update-ignore
+++ b/update-ignore.d/01update-ignore
@@ -126,7 +126,14 @@ if [ -e "$file" ]; then
 		exit 1
 	fi
 	realfile="$file"
-	file=$(tempfile)
+	if [ -n "`type -p tempfile`" ]; then
+		tempfile="tempfile"
+	elif [ -n "`type -p mktemp`" ]; then
+		tempfile="mktemp"
+	else
+		echo "etckeeper warning: can't find tempfile or mktemp" >&2
+	fi
+	file=$($tempfile)
 	(
 		skipping=
 		while read line; do
